Buzz-pbox/patch-1

pboxx opened the pull request:
- Incorporated Inge's latest wrappers (refactored to e.g. meet naming conventions).
- Replaced ushort tag values and hard-coded enums with a parsing functionality of the EAxian XML that contains the catalog categories and tags. Tags are deserialized into a custom class which is used instead of the ushort/values values. Also included is a custom editor to edit the value in the grid.
- Edited OBJD to correctly read/display EmotionEnvironmentTag indexes
- Added XML resource with tag information to project.

Kuree commented:
Why to change the entire implementation?

BrutalBuzzler commented:
Not sure which implementation you mean; flags or wrappers. Inge's wrappers are more complete regarding the semantic information like "PlacementFlags" vs. Unknown or something. pbox and I refactored them to be a little more in line with the usual C# conventions, though. The tag thing is because the previous implementation was a PITA (I was told) to use, because you had a dropdown with 1677 elements to choose from. It was also hard-coded while now you just need to drop the new resource into the S4pe folder and have the most current reincarnation of the EAxian tags.

BTW: There was an issue with the output directory of the new project in the first version of this request. I just fixed it and pbox updated the pull request. The project references of the solution seem to be a mess, though, with hard references to the compiled assemblies instead of the projects. That's why the output directory of each project even matters. I may have a look at that at some point.

Kuree commented:
The pull-request changed the way that most catalog resources behave. The current implementation has self-clone feature.

pboxx commented:
Changing the way that the catalog resources behave is exactly the point of this. 

Regarding the clone feature, do you mean the Alex Burtsev stuff that is used in COBJ? If so: what is the actual advantage of that? It would probably not be a big deal to move that over to Inge’s version, I just don’t know of anything that doesn’t work in hers.

cmarNYC committed:
Create ImportJFIF.resx

cmarNYC committed:
Create JFIFImportHelper.csproj

cmarNYC committed:
Create Program.cs

cmarNYC committed:
Create ThumbnailImport.helper

cmarNYC committed:
Create AssemblyInfo.cs

cmarNYC committed:
Create Resources.Designer.cs

cmarNYC committed:
Create Resources.resx

cmarNYC committed:
Create Settings.Designer.cs

cmarNYC committed:
Create Settings.settings

cmarNYC committed:
Create DMAPImageExport.helper

cmarNYC committed:
Create DMAPImageExportRobe.helper

cmarNYC committed:
Create DMAPImageHelper.csproj

cmarNYC committed:
Create Program.cs

cmarNYC committed:
Create AssemblyInfo.cs

cmarNYC committed:
Create Resources.Designer.cs

cmarNYC committed:
Create Resources.resx

cmarNYC committed:
Create Settings.Designer.cs

cmarNYC committed:
Create Settings.settings

cmarNYC committed:
Display of more useful information

Image export corrected and available as a resource helper

cmarNYC committed:
Update CatalogCommon.cs

cmarNYC committed:
Added backwards compatibility of property tags

cmarNYC committed:
Updated with info from EA bt resource description

cmarNYC committed:
Added constructor to support backwards compatibility

cmarNYC committed:
Fixed and updated

Fixed to accept STBLs with incorrect (non-EA) total string length generated by TSRW (works fine in-game), updated to include flags for each string

BrutalBuzzler committed:
Fix update checking, make visiting update site optional and fix some headers

* Update checking now directs to the s4ptacle release page and supports more complicated version strings.
* The dialog that is shown when a newer version is found on the release page, now asks the user whether to visit the download page.
* Updated the copyright headers of the touched files to reflect contributors.

cmarNYC committed:
Delete CLIP exportAsNewName.csproj

cmarNYC committed:
Delete CLIP exportAsNewName.sln

cmarNYC committed:
Delete CLIPexportAsNewName.helper

cmarNYC committed:
Delete MainForm.Designer.cs

cmarNYC committed:
Delete MainForm.cs

cmarNYC committed:
Delete MainForm.resx

cmarNYC committed:
Delete Program.cs

cmarNYC committed:
Delete AssemblyInfo.cs

cmarNYC committed:
Delete AssemblyVersion.cs

cmarNYC committed:
Delete Settings.Designer.cs

cmarNYC committed:
Delete Settings.settings

cmarNYC committed:
Delete MainForm.Designer.cs

cmarNYC committed:
Delete MainForm.cs

cmarNYC committed:
Delete MainForm.resx

cmarNYC committed:
Delete OBJK Resource Editor.csproj

cmarNYC committed:
Delete OBJK Resource Editor.sln

cmarNYC committed:
Delete OBJKResourceEditor.helper

cmarNYC committed:
Delete Program.cs

cmarNYC committed:
Delete AssemblyInfo.cs

cmarNYC committed:
Delete AssemblyVersion.cs

cmarNYC committed:
Delete Resources.Designer.cs

cmarNYC committed:
Delete Resources.resx

cmarNYC committed:
Delete s3pe.ico

cmarNYC committed:
Delete Export.Designer.cs

cmarNYC committed:
Delete Export.cs

cmarNYC committed:
Delete Export.resx

cmarNYC committed:
Delete Import.Designer.cs

cmarNYC committed:
Delete Import.cs

cmarNYC committed:
Delete Import.resx

cmarNYC committed:
Delete MyProgressBar.cs

cmarNYC committed:
Delete Program.cs

cmarNYC committed:
Delete S3SA DLL ExpImp.csproj

cmarNYC committed:
Delete S3SA DLL ExpImp.sln

cmarNYC committed:
Delete S3SADLLExport.helper

cmarNYC committed:
Delete S3SADLLImport.helper

cmarNYC committed:
Delete S3SADLLViewer_Go.helper

cmarNYC committed:
Delete S3SADLLViewer_Setup.helper

cmarNYC committed:
Delete View.Designer.cs

cmarNYC committed:
Delete View.cs

cmarNYC committed:
Delete View.resx

cmarNYC committed:
Delete AssemblyInfo.cs

cmarNYC committed:
Delete AssemblyVersion.cs

cmarNYC committed:
Delete Settings.Designer.cs

cmarNYC committed:
Delete Settings.settings

cmarNYC committed:
Delete App.config

cmarNYC committed:
Delete MainForm.Designer.cs

cmarNYC committed:
Delete MainForm.cs

cmarNYC committed:
Delete MainForm.resx

cmarNYC committed:
Delete Program.cs

cmarNYC committed:
Delete STBL Resource Editor.csproj

cmarNYC committed:
Delete STBL Resource Editor.sln

cmarNYC committed:
Delete STBLResourceEditor.helper

cmarNYC committed:
Delete AssemblyInfo.cs

cmarNYC committed:
Delete AssemblyVersion.cs

cmarNYC committed:
Delete Resources.Designer.cs

cmarNYC committed:
Delete Resources.resx

cmarNYC committed:
Delete s3pe.ico

cmarNYC committed:
Delete Program.cs

cmarNYC committed:
Delete VP6-to-_VID ExpImp.csproj

cmarNYC committed:
Delete VP6-to-_VID ExpImp.sln

cmarNYC committed:
Delete VideoResource.cs

cmarNYC committed:
Delete AssemblyInfo.cs

cmarNYC committed:
Delete AssemblyVersion.cs

cmarNYC committed:
Delete MainForm.Designer.cs

cmarNYC committed:
Delete MainForm.cs

cmarNYC committed:
Delete MainForm.resx

cmarNYC committed:
Delete Program.cs

cmarNYC committed:
Delete VPXY Resource Editor.csproj

cmarNYC committed:
Delete VPXY Resource Editor.sln

cmarNYC committed:
Delete VPXYResourceEditor.helper

cmarNYC committed:
Delete AssemblyInfo.cs

cmarNYC committed:
Delete AssemblyVersion.cs

cmarNYC committed:
Delete Resources.Designer.cs

cmarNYC committed:
Delete Resources.resx

cmarNYC committed:
Delete s3pe.ico

cmarNYC committed:
Delete Packages.dgml

cmarNYC committed:
Delete ViewDDS.sln

cmarNYC committed:
Delete gpl-3.0.txt

cmarNYC committed:
Delete makedist-DDSTool.bat

cmarNYC committed:
Delete mknsis.nsi

cmarNYC committed:
Delete DDSTool.csproj

cmarNYC committed:
Delete MainForm.Designer.cs

cmarNYC committed:
Delete MainForm.cs

cmarNYC committed:
Delete MainForm.resx

cmarNYC committed:
Delete NewDDSParameters.Designer.cs

cmarNYC committed:
Delete NewDDSParameters.cs

cmarNYC committed:
Delete NewDDSParameters.resx

cmarNYC committed:
Delete Program.cs

cmarNYC committed:
Delete s3peDDSTool.helper

cmarNYC committed:
Delete AssemblyInfo.cs

cmarNYC committed:
Delete AssemblyVersion.cs

cmarNYC committed:
Delete Resources.Designer.cs

cmarNYC committed:
Delete Resources.resx

cmarNYC committed:
Delete MainForm.Designer.cs

cmarNYC committed:
Delete MainForm.cs

cmarNYC committed:
Delete MainForm.resx

cmarNYC committed:
Delete Program.cs

cmarNYC committed:
Delete ViewDDS.csproj

cmarNYC committed:
Delete sim3tools_ViewDDS.helper

cmarNYC committed:
Delete AssemblyInfo.cs

cmarNYC committed:
Delete ImportJFIF.Designer.cs

cmarNYC committed:
Delete ImportJFIF.cs

cmarNYC committed:
Delete ImportJFIF.resx

cmarNYC committed:
Delete JFIFImportHelper.csproj

cmarNYC committed:
Delete Program.cs

cmarNYC committed:
Delete ThumbnailImport.helper

cmarNYC committed:
Delete AssemblyInfo.cs

cmarNYC committed:
Delete Resources.Designer.cs

cmarNYC committed:
Delete Resources.resx

cmarNYC committed:
Delete Settings.Designer.cs

cmarNYC committed:
Delete Settings.settings

cmarNYC committed:
Create DDSExport.helper

cmarNYC committed:
Create DDSHelper.csproj

cmarNYC committed:
Create DDSImport.helper

cmarNYC committed:
Create Import.cs

cmarNYC committed:
Create Import.Designer.cs

cmarNYC committed:
Create Import.resx

cmarNYC committed:
Create Program.cs

cmarNYC committed:
Create AssemblyInfo.cs

cmarNYC committed:
Create Resources.Designer.cs

cmarNYC committed:
Create Resources.resx

cmarNYC committed:
Create Settings.Designer.cs

cmarNYC committed:
Create Settings.settings

cmarNYC committed:
Update DMAPImageExport.helper

cmarNYC committed:
Update DMAPImageExportRobe.helper

cmarNYC committed:
Changed output path

cmarNYC committed:
Update of credits

cmarNYC committed:
Name changes

cmarNYC committed:
Update GEOMViewer.helper

cmarNYC committed:
Changed output path

cmarNYC committed:
Changed output path

cmarNYC committed:
Changed output path

cmarNYC committed:
Changed output path

cmarNYC committed:
Changed output path

cmarNYC committed:
Changed output path

cmarNYC committed:
Cosmetic changes

cmarNYC committed:
Cosmetic changes

cmarNYC committed:
Changes to credits, cosmetics

cmarNYC committed:
Changes to credits, cosmetics

cmarNYC committed:
Changed output path

cmarNYC committed:
Update RLESImport.helper

cmarNYC committed:
Changed output path and names

cmarNYC committed:
Fixed names

cmarNYC committed:
Create Settings.cs

cmarNYC committed:
Create ImportThumb.cs

cmarNYC committed:
Create ImportThumb.Designer.cs

cmarNYC committed:
Create ImportThumb.resx

cmarNYC committed:
Create Program.cs

cmarNYC committed:
Create ThumbnailExport.helper

cmarNYC committed:
Create ThumbnailHelper.csproj

cmarNYC committed:
Create ThumbnailImport.helper

cmarNYC committed:
Create AssemblyInfo.cs

cmarNYC committed:
Create Resources.Designer.cs

cmarNYC committed:
Create Resources.resx

cmarNYC committed:
Create Settings.Designer.cs

cmarNYC committed:
Create Settings.settings

cmarNYC committed:
Added helper projects

cmarNYC committed:
Fixed comparer method

cmarNYC committed:
Fixed Grid editing

cmarNYC committed:
Include System.Custom reference

cmarNYC committed:
Delete outdated and redundant DMAP wrapper

cmarNYC committed:
Create SkyBoxTextureResource.cs

cmarNYC committed:
Create WorldColorTimelineResource.cs

cmarNYC committed:
Remove 1 and add 2 wrappers

Removed DefomerMapResource, added SkyBoxTextureResource and WorldColorTimelineResource

bstenzel committed:
Clean up Clip, ClipResource and AApiVersionedFields

cmarNYC committed:
Add missing tags and extensions

Contributed by Inge Jones.

cmarNYC committed:
Fix duplicate, add DMAP and Skybox Color

BrutalBuzzler committed:
Revert "Clean up Clip, ClipResource and AApiVersionedFields"

This reverts commit 70522806dfa07b93ac96dc8d2c108145de799e38.

cmarNYC committed:
Fix in CASPartResource flags, credits updated

BrutalBuzzler committed:
Add support for import of resource in the TS4 naming convention

BrutalBuzzler committed:
Refactor damage done to the tags on making the related resources backwards compatible

BrutalBuzzler committed:
Attempt to fix build (WIP)

BrutalBuzzler committed:
Add readonly option to file open dialog and apply style conventions to MainForm

BrutalBuzzler committed:
Fix StblEntry so it can be added and cloned in the grid again

BrutalBuzzler committed:
Fix adding new STBL resources (by Inge)

BrutalBuzzler committed:
Extract superclass from COBJResource and make CCOLResource inherit from it (thus fixing CCOL)

BrutalBuzzler committed:
Clean up MainForm and fix bookmarks dialog

BrutalBuzzler committed:
Update extensions mapping (by Inge)

cmarNYC committed:
Additional options in DMAP export

cmarNYC committed:
Update of CLIP wrapper

Uses info provided by EA

cmarNYC committed:
Merge branch 'develop' of https://github.com/s4ptacle/Sims4Tools into develop

# Conflicts:
#	s4pi Wrappers/AnimationResources/ClipResource.cs

BrutalBuzzler committed:
Make each helper build into its own folder again (should fix concurrency issues during build too)

pboxx committed:
Add TRIMResource

Adds a wrapper for TRIM 0x76BCF80C.

pboxx committed:
Update tag tuning XML

Tag tuning from game version 1.15 (2016-02-04)

cmarNYC committed:
Added missing lines for DDS export

cmarNYC committed:
Merge branch 'develop' of https://github.com/s4ptacle/Sims4Tools into develop

# Conflicts:
#	s4pe Helpers/DDSHelper/DDSExport.helper

cmarNYC committed:
Update helper folder location

pboxx committed:
Update README.md

pboxx committed:
Update README.md

pboxx committed:
Update README.md

pboxx committed:
Update README.md

pboxx committed:
Update README.md

pboxx committed:
Update README.md

BrutalBuzzler committed:
Fix AbstractCatalogResource (COBJ and CCOL) for version 1.15

Made resource only use ambienceFileInstanceId and isOverrideInstance for exactly auralPropertiesVersion 3 and added a new unknown01 byte for version 4. This seems to work for now. The following properties are in order again now. This may still be completely wrong as the parts we believe to be unused may be simply be used for different purposes now.

BrutalBuzzler committed:
Merge branch 'develop' of https://github.com/s4ptacle/Sims4Tools into develop

BrutalBuzzler committed:
CatalogCommon is now version 0xB by default and COBJ and CCOL can be added to packages (fallbackObjectKey will be instantiated as empty TGIBlock instead of being left null)

pboxx committed:
Update Extensions.txt

Update with new and corrected file types on behalf of Inge.

pboxx committed:
Add png import/export for THUM 0x16CCF748, 0xE18CAEE2, 0xE254AE6E

pboxx committed:
Update MTBL

Updated with info from the EAxian .bt

pboxx committed:
Update MTBL, correct Extensions.txt

pboxx committed:
Update MTBL

Update with additional flag lists from EA template update 2012-02-12

BrutalBuzzler committed:
Add new Open Readonly toolstrip menu item, because "ShowReadOnly" makes the file open dialog show an entirely different dialog

pboxx committed:
Update various headers

pboxx committed:
Merge branch 'develop' of https://github.com/s4ptacle/Sims4Tools into develop

cmarNYC committed:
Fixes to CLIP wrapper

Fixed bugs in write, in read/write of packed frame indices, fixed addition of empty CLIP and CLIP parts.

cmarNYC committed:
Merge branch 'develop' of https://github.com/s4ptacle/Sims4Tools into develop

pboxx committed:
Update README.md

pboxx committed:
Update README.md

pboxx committed:
Update header

pboxx committed:
Merge branch 'develop' of https://github.com/s4ptacle/Sims4Tools into develop

cmarNYC committed:
Correct index array order in Frame

Fixed reversed index order in one type of Frame

pboxx committed:
Update Extensions.txt with all known XML types as of 1.15

pboxx committed:
Update some headers

pboxx committed:
Merge branch 'develop' of https://github.com/s4ptacle/Sims4Tools into develop

pboxx committed:
Update README.md

pboxx committed:
Update TextResources.txt, Extensions.txt

Mainly removing all the TS3 types, also adding a few missing ones

pboxx committed:
Merge branch 'develop' of https://github.com/s4ptacle/Sims4Tools into develop

pboxx committed:
Update TextResources.txt

pboxx committed:
Update Extensions.txt (again)

BrutalBuzzler committed:
Reformat ExtList and throw more meaningful exception on duplicate entries

pboxx committed:
Update for 0.4.3

pboxx committed:
Merge branch 'release/0.4.3'

Conflicts:
	README.md

pboxx committed:
Fix old link in readme

pboxx committed:
Update README.md

Add link to 0.4.4-beta

cmarNYC committed:
Update README.md
